versioning/scripts/dockerfiles/main.go

```diff
@@ -173,30 +180,39 @@ func check(e error) {
 func main() {
 	templateDirPtr := flag.String("template_dir", "templates", "Path to directory containing Dockerfile.template and any other files to copy over")
 	verifyPtr := flag.Bool("verify_only", false, "Verify dockerfiles")
+	var templatePath string
+	var version = "0.01"
+
+	log.Printf("dockerfiles verson: %s", version)
+
 	flag.Parse()
 
 	var spec versions.Spec
 	spec = versions.LoadVersions("versions.yaml")
 
-	templatePath := filepath.Join(*templateDirPtr, "Dockerfile.template")
-	templateData, err := ioutil.ReadFile(templatePath)
-	templateString := string(templateData)
-	check(err)
+	for _, version := range spec.Versions {
+		if version.TemplateDir != "" {
+			templatePath = filepath.Join(*templateDirPtr, version.TemplateDir, "Dockerfile.template")
+		} else {
+			templatePath = filepath.Join(*templateDirPtr, "Dockerfile.template")
+		}
+		templateData, err := ioutil.ReadFile(templatePath)
+		templateString := string(templateData)
+		check(err)
 
-	tmpl, err := template.
-		New("dockerfileTemplate").
-		Parse(templateString)
-	check(err)
+		tmpl, err := template.
+			New("dockerfileTemplate").
+			Parse(templateString)
+		check(err)
 
-	if *verifyPtr {
-		failureCount := verifyDockerfiles(spec, *tmpl)
-		failureCount += verifyCopiedFiles(spec, *templateDirPtr)
-		os.Exit(failureCount)
-	} else {
-		for _, version := range spec.Versions {
+		if *verifyPtr {
+			failureCount := verifyDockerfiles(spec, *tmpl)
+			failureCount += verifyCopiedFiles(spec, *templateDirPtr)
+			os.Exit(failureCount)
+		} else {
 			data := renderDockerfile(version, *tmpl)
 			writeDockerfile(version, data)
-			copyFiles(version, *templateDirPtr)
+			copyFiles(version, templatePath)
```

Comment:
`templateDir`

Reply:
Here we should use the variable that points to subfolder storing Dockerfile.template - in this case this variable is templatePath.

Reply:
If I read the code correctly, `templatePath` is actually the path to the Dockerfile template. This is why in my other comment I suggested adding another variable `templateDir` for this method.

Reply:
Good catch! I corrected the issue.